qt/aqt/deckoptions.py

```diff
@@ -57,15 +60,58 @@ def _setup_ui(self) -> None:
 
     def _on_bridge_cmd(self, cmd: str) -> None:
         if cmd == "deckOptionsReady":
+            self.ready = True
             gui_hooks.deck_options_did_load(self)
 
+    def closeEvent(self, evt: QCloseEvent) -> None:
+        if self._close_event_has_cleaned_up:
+            evt.accept()
+            return
+        evt.ignore()
+        self.ifCanClose()
+
+    def _close(self):
+        """Close. Ensure the closeEvent is not ignored."""
+        self._close_event_has_cleaned_up = True
+        self.close()
+
+    def ifCanClose(self):
+        """Close if there was no modification. Otherwise ask for confirmation first."""
+
+        def callbackWithUserChoice(choice: int):
+            if choice == 0:
+                # The user accepted to discard current input.
+                self._close()
+
+        def callbackWithDataInformation(hasModifiedData: bool):
+            if hasModifiedData:
+                ask_user_dialog(
+                    tr.adding_discard_current_input(),
+                    callback=callbackWithUserChoice,
+                    buttons=[
+                        QMessageBox.StandardButton.Discard,
+                        (tr.adding_keep_editing(), QMessageBox.ButtonRole.RejectRole),
+                    ],
+                )
+            else:
+                self._close()
+
+        self.hasModifiedData(callbackWithDataInformation)
+
     def reject(self) -> None:
         self.mw.col.set_wants_abort()
         self.web.cleanup()
         self.web = None
         saveGeom(self, self.TITLE)
         QDialog.reject(self)
 
+    def hasModifiedData(self, callback: Callable[[bool], None]):
+        """Calls `callback` with the information of whether any deck options are modified."""
+        if self.ready:
+            self.web.evalWithCallback("globalThis.isStateModified()", callback)
```

Comment:
This is not a great name - it's not specific to the deck options screen, and may conflict with other usages. We also don't want to be polluting the global namespace like this. This is better placed until a prefix for Anki, and for the deck options, like 'anki.deckOptionsPendingChanges()'

Reply:
How do I create such a prefix?
If I try to directly do 
```
anki.deckOptionsPendingChanges = () => {
        return data.state.isModified();
    };
```

I got an error message telling me that `anki` is not defined.

My understanding is that I can't just declare it in the svelte file otherwise it would not be accessible from python. If I try to declare `export val anki` in `+page.ts` i've got the same problem. This is why I thought using the `globalThis` would be the only way to have a value in svelte that could escape svelte.

Reply:
An example from another file:

```typescript
    globalThis.anki = globalThis.anki || {};
    globalThis.anki.imageOcclusion = {
        mode,
        save,
    };
```

Reply:
oh okay, `anki` is a field of `globalThis`. Nice